llmfoundry/utils/exceptions.py

```diff
@@ -387,3 +387,48 @@ class RunTimeoutError(InternalError):
     def __init__(self, timeout: int) -> None:
         message = f'Run timed out after {timeout} seconds.'
         super().__init__(message, timeout=timeout)
+
+
+class LossSpikeError(UserError):
+    """Error thrown a severe loss spike occurs."""
+
+    def __init__(
+        self,
+        outlier_multiplier: float,
+        running_loss_avg: int,
+        outlier_counter: int,
+    ) -> None:
+        if outlier_multiplier and running_loss_avg and outlier_counter:
+            message = f'Training stopped due to a loss spike. The training loss was more than {outlier_multiplier} times greater than \
+                        the running average loss (approx. {running_loss_avg}) over {outlier_counter} consecutive training steps. \
+                        Please try submitting the run again with a lower learning rate.'
+
+        else:
+            message = 'Training stopped due to a loss spike. Please try submitting the run again with a lower learning rate.'
+        super().__init__(
+            message,
+            outlier_multiplier=outlier_multiplier,
+            running_loss_avg=running_loss_avg,
+            outlier_counter=outlier_counter,
+        )
+
+
+class HighLossError(UserError):
+    """Error thrown training loss plateaus or is unstable at a high level."""
+
+    def __init__(
+        self,
+        loss_cap: float,
+        window_size: int,
+    ) -> None:
+        if loss_cap and window_size:
+            message = f'Training stopped due to consistently high losses. The training loss exceeded the threshold of {loss_cap} \
+                        for more than half of the {window_size} most recent training steps. Please try submitting the run again with a lower learning rate.'
+
+        else:
```

Comment:
why is the else needed? it looks to me like it would never get triggered

Reply:
yeah, it was more just in case the error would be raised elsewhere in future without the same info, but removed these now